From d598fa8b3129870d323fbaa889ab389d28526a1b Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Tue, 31 Dec 2024 17:23:35 +0100
Subject: [PATCH 2/2] test: Skip shake* tests on OpenSSL 3.4

shake* requires a length to be set.
Skip the tests for now.

=> https://github.com/nodejs/node/issues/56159
=> https://github.com/openssl/openssl/commit/b911fef216d1386210ec24e201d54d709528abb4

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 test/parallel/test-crypto-hash.js         | 3 +++
 test/parallel/test-crypto-oneshot-hash.js | 3 +++
 2 files changed, 6 insertions(+)

--- a/test/parallel/test-crypto-hash.js
+++ b/test/parallel/test-crypto-hash.js
@@ -182,6 +182,8 @@
 
 // Test XOF hash functions and the outputLength option.
 {
+  // Skip shake XOF due to length requirement.
+  if (!common.hasOpenSSL(3, 4)) {
   // Default outputLengths.
   assert.strictEqual(crypto.createHash('shake128').digest('hex'),
                      '7f9c2ba4e88f827d616045507605853e');
@@ -195,6 +197,7 @@
                            .digest('hex'),
                      '46b9dd2b0ba88d13233b3feb743eeb24' +
                      '3fcd52ea62b81b82b50c27646ed5762f');
+  }
 
   // Short outputLengths.
   assert.strictEqual(crypto.createHash('shake128', { outputLength: 0 })
--- a/test/parallel/test-crypto-oneshot-hash.js
+++ b/test/parallel/test-crypto-oneshot-hash.js
@@ -31,6 +31,9 @@
 const input = fs.readFileSync(fixtures.path('utf8_test_text.txt'));
 
 for (const method of methods) {
+  // Skip shake XOF due to length requirement.
+  if (common.hasOpenSSL(3, 4) && (method === 'shake128') || (method === 'shake256'))
+  continue;
   for (const outputEncoding of ['buffer', 'hex', 'base64', undefined]) {
     const oldDigest = crypto.createHash(method).update(input).digest(outputEncoding || 'hex');
     const digestFromBuffer = crypto.hash(method, input, outputEncoding);
